### PR TITLE
[common,types] Rename NodeUnitConfig

### PR DESCRIPTION
**Pull request by mlohvynenko**

### CODE REVIEW

File: include/aos/sm/resourcemanager.hpp [hunk 1]
```diff
@@ -320,7 +320,7 @@ class ResourceManager : public ResourceManagerItf, private NonCopyable {
     static constexpr auto cConfigJSONLen = AOS_CONFIG_UNIT_CONFIG_JSON_LEN;
 
     Error LoadUnitConfig();
-    Error ValidateUnitConfig(const NodeUnitConfig& nodeUnitConfig) const;
+    Error ValidateUnitConfig(const NodeConfig& nodeConfig) const;
```
Comment (al1img) [inline include/aos/sm/resourcemanager.hpp:323]: `ValidateNodeConfig`?

File: include/aos/sm/resourcemanager.hpp [hunk 2]
```diff
@@ -22,7 +22,7 @@ namespace resourcemanager {
  * Unit config contains single node unit details.
  */
 struct UnitConfig {
-    NodeUnitConfig            mNodeUnitConfig;
+    NodeConfig                mNodeConfig;
```
Comment (al1img) [inline include/aos/sm/resourcemanager.hpp:25]: This is also node config but with version. And as they are in different namespaces, we can name this structure as `NodeConfig` also?

File: include/aos/sm/resourcemanager.hpp [hunk 3]
```diff
@@ -320,7 +320,7 @@ class ResourceManager : public ResourceManagerItf, private NonCopyable {
     static constexpr auto cConfigJSONLen = AOS_CONFIG_UNIT_CONFIG_JSON_LEN;
 
     Error LoadUnitConfig();
-    Error ValidateUnitConfig(const NodeUnitConfig& nodeUnitConfig) const;
+    Error ValidateUnitConfig(const NodeConfig& nodeConfig) const;
     Error ValidateDevices(const Array<DeviceInfo>& devices) const;
     Error GetUnitConfigDeviceInfo(const String& deviceName, DeviceInfo& deviceInfo) const;
```
Comment (al1img) [inline include/aos/sm/resourcemanager.hpp:325]: GetUnitConfigDeviceInfo -> GetNodeConfigDeviceInfo

File: include/aos/sm/resourcemanager.hpp [hunk 4]
```diff
@@ -39,20 +39,20 @@ class JSONProviderItf {
     /**
      * Dumps node unit config object from json string.
      *
-     * @param nodeUnitConfig node unit config object.
+     * @param nodeConfig node config object.
      * @param[out] json json representation of node unit config.
      * @return Error.
      */
-    virtual Error DumpUnitConfig(const UnitConfig& nodeUnitConfig, String& json) const = 0;
+    virtual Error DumpUnitConfig(const UnitConfig& nodeConfig, String& json) const = 0;
```
Comment (al1img) [inline include/aos/sm/resourcemanager.hpp:46]: DumpNodeConfig

File: include/aos/sm/resourcemanager.hpp [hunk 5]
```diff
@@ -39,20 +39,20 @@ class JSONProviderItf {
...
     /**
      * Parses node unit config json string from object.
      *
      * @param json json representation of node unit config.
-     * @param[out] nodeUnitConfig node unit config.
+     * @param[out] nodeConfig node unit config.
      * @return Error.
      */
-    virtual Error ParseNodeUnitConfig(const String& json, UnitConfig& nodeUnitConfig) const = 0;
+    virtual Error ParseNodeUnitConfig(const String& json, UnitConfig& nodeConfig) const = 0;
```
Comment (al1img) [inline include/aos/sm/resourcemanager.hpp:55]: ParseNodeConfig

File: include/aos/sm/resourcemanager.hpp [hunk 6]
```diff
@@ -37,22 +37,22 @@ class JSONProviderItf {
...
+     * Dumps config object into string.
      *
-     * @param nodeConfig node config object.
-     * @param[out] json json representation of node unit config.
+     * @param config config object.
+     * @param[out] json json representation of config.
      * @return Error.
      */
-    virtual Error DumpUnitConfig(const UnitConfig& nodeConfig, String& json) const = 0;
+    virtual Error DumpConfig(const Config& config, String& json) const = 0;
```
Comment (al1img) [inline include/aos/sm/resourcemanager.hpp:46]: DumpNodeConfig

File: include/aos/sm/resourcemanager.hpp [hunk 7]
```diff
@@ -37,22 +37,22 @@ class JSONProviderItf {
...
+     * Parses config object from string.
      *
-     * @param json json representation of node unit config.
-     * @param[out] nodeConfig node unit config.
+     * @param json json representation of config.
+     * @param[out] config config.
      * @return Error.
      */
-    virtual Error ParseNodeUnitConfig(const String& json, UnitConfig& nodeConfig) const = 0;
+    virtual Error ParseConfig(const String& json, Config& config) const = 0;
```
Comment (al1img) [inline include/aos/sm/resourcemanager.hpp:55]: ParseNodeConfig

File: include/aos/sm/resourcemanager.hpp [hunk 8]
```diff
@@ -201,22 +201,22 @@ class ResourceManagerItf {
...
+     * Checks configuration.
      *
-     * @param version unit config version
-     * @param unitConfig string with unit configuration.
+     * @param version config version
+     * @param config string with configuration.
      * @return Error.
      */
-    virtual Error CheckUnitConfig(const String& version, const String& unitConfig) const = 0;
+    virtual Error CheckConfig(const String& version, const String& config) const = 0;
```
Comment (al1img) [inline include/aos/sm/resourcemanager.hpp:210]: CheckNodeConfig

File: include/aos/sm/resourcemanager.hpp [hunk 9]
```diff
@@ -201,22 +201,22 @@ class ResourceManagerItf {
...
+     * Updates configuration.
      *
-     * @param version unit config version.
-     * @param unitConfig string with unit configuration.
+     * @param version config version.
+     * @param config string with configuration.
      * @return Error.
      */
-    virtual Error UpdateUnitConfig(const String& version, const String& unitConfig) = 0;
+    virtual Error UpdateConfig(const String& version, const String& config) = 0;
```
Comment (al1img) [inline include/aos/sm/resourcemanager.hpp:219]: UpdateNodeConfig

File: include/aos/sm/resourcemanager.hpp (hunk 1 above)
Comment (mlohvynenko) [reply include/aos/sm/resourcemanager.hpp]: In this commit I've renamed NodeUnitConfig in the common
In the next commit I've renamed sm NodeConfig struct and updated method names.

File: include/aos/sm/resourcemanager.hpp (hunk 2 above)
Comment (mlohvynenko) [reply include/aos/sm/resourcemanager.hpp]: JSON provider uses this struct in interface methods.
This sturct is dumped/parsed into/from JSON.

**Review by al1img — APPROVED** [review]

**Pull request merged**